### PR TITLE
fix: dont omit signoptions in sign and signasync

### LINKED ISSUE #1369: type error while using `expiresIn` in `signAsync`
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

I am not able to use `expiresIn` option while signing a token

Getting error

```
No overload matches this call.
  Overload 1 of 2, '(payload: string, options?: Omit<JwtSignOptions, keyof SignOptions>): Promise<string>', gave the following error.
    Argument of type '{ secret: string; expiresIn: string; }' is not assignable to parameter of type 'Omit<JwtSignOptions, keyof SignOptions>'.
      Object literal may only specify known properties, and 'expiresIn' does not exist in type 'Omit<JwtSignOptions, keyof SignOptions>'.
  Overload 2 of 2, '(payload: object | Buffer, options?: JwtSignOptions): Promise<string>', gave the following error.
    Argument of type 'string' is not assignable to parameter of type 'object | Buffer'.ts(2769)
```

In following code

```typescript
const accessToken = this.jwtService.signAsync(String(user.id), {
     secret: accessTokenPublicKey,
     expiresIn: "15m",
});
```

Code for module

```typescript
@Module({
	controllers: [AuthController],
	providers: [AuthService],
	imports: [JwtModule.register({ global: true, signOptions: { algorithm: "RS256" } })],
})
export class AuthModule {}
```

I can use `expiresIn` in `JwtModule.register()` but now while using it in service

### Minimum reproduction code

https://github.com/maya-manager/server/blob/aayushchugh/login/src/modules/auth/auth.service.ts#L164-L167

### Steps to reproduce

_No response_

### Expected behavior

I should be able to use `expiresIn` without any type errors

### Package version

10.1.0

### NestJS version

10.0.5

### Node.js version

18.15.0

### In which operating systems have you tested?

- [X] macOS
- [ ] Windows
- [ ] Linux

### Other

image

Also, VScode is only showing autocomplete for these 2 options

image

Here there are two `signAsync` and in the upper one you are using `Omit`. I think `Omit` is removing all the types from `SignOptions` and I can't use `expiresIn`

image

### PR DESCRIPTION
**Pull request by aayushchugh**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #1369 
fixes #1369 

## What is the new behavior?
1. Removed `Omit` type from `sign` and `signAsync`
2. Failing tests are now passing
3. Removed unused `@ts-expect-error`
4. Allowed string type as parameter in `sign` and `signAsync` types

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

image

### CODE REVIEW

File: lib/jwt.service.spec.ts [hunk 1]
```diff
@@ -340,16 +340,14 @@ describe('JWT Service', () => {
 
     it('should "sign" expect errors with a "payload" string with "expiresIn"', () => {
       expect(() =>
-        // @ts-expect-error
-        jwtService.sign(testPayloadStr, { expiresIn: 60 })
+        jwtService.sign(testPayloadStr, { expiresIn: 60 } as jwt.SignOptions)
```
Comment (kamilmysliwiec) [inline lib/jwt.service.spec.ts:343]: cc @Hender-hs

Comment (aayushchugh) [reply lib/jwt.service.spec.ts]: removed `as jwt.SignOptions`

**Review by tianangthang7 — APPROVED** [review]
nice job

**Comment by aayushchugh** [comment]
@kamilmysliwiec Any update on this PR?

**Comment by aayushchugh** [comment]
Closing this PR

**Pull request closed without merging**